### DISCUSSION TITLE
Does the tink-worker support containers in Host NetworkMode for Tinker Actions?

### DISCUSSION THREAD

**Discussion by vthiruveedula** [tinkerbell/tink#1037] [Q&A]
Hello Team,

One of the Tinker action in our tinker workflow needs to communicate with services running on the host machine's localhost on a specific port. 
To achieve this, I believe the container running this action needs to be executed in Host NetworkMode.
Could anyone please confirm if the tink-worker supports scheduling containers in Host NetworkMode for Tinker actions? 
If so, could you provide guidance or examples on how to configure this?

Thank you for your assistance!

Best Regards,
Vijaya

**Comment by jacobweinstock** [comment]
Hey @vthiruveedula . In our current [v1alpha1 API ](https://github.com/tinkerbell/tink/tree/main/api/v1alpha1) specifying a network namespace is not directly supported in the spec. In the unimplemented [v1alpha2 API](https://github.com/tinkerbell/tink/tree/main/api/v1alpha2) it will be directly supported.

With that said, there is at least one way to make the host network available to an Action. When you define your Action, make sure `pid: host` is set. Then in the Action itself, run a command to enter the host network namespace. For example, `nsenter -t1 -n sh`. Then run the Action.

Hopefully that helps. Let me know if i can elaborate on anything. Thanks for posting!

**Reply by vthiruveedula** [reply]
Yes @jacobweinstock . That helps. Thanks!
